### ISSUE TITLE
Release test air_example_gptj_deepspeed_fine_tuning failed

### ISSUE THREAD

**Issue by can-anyscale** [ray-project/ray#41391]
Release test **air_example_gptj_deepspeed_fine_tuning** failed. See https://buildkite.com/ray-project/release/builds/2173#018c0eac-19be-4597-9712-7583322cf110 for more details.

Managed by OSS Test Policy

**Comment by justinvyu** [comment]
This looks like the same error as https://github.com/ray-project/ray/issues/40547, which was fixed by adding a seeded random shuffle on the data: https://github.com/ray-project/ray/pull/40648

```
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/utils/nvtx.py", line 15, in wrapped_fn
    ret_val = func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/zero/stage3.py", line 1896, in _overflow_check_and_loss_scale_update
    self._update_scale(self.overflow)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/zero/stage3.py", line 2249, in _update_scale
    self.loss_scaler.update_scale(has_overflow)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/fp16/loss_scaler.py", line 175, in update_scale
    raise Exception(
Exception: Current loss scale already at minimum - cannot decrease scale anymore. Exiting run.
```

**Comment by can-anyscale** [comment]
Likely still failing, I don't see a fix yet

**Comment by can-anyscale** [comment]
Test has been failing for far too long. Jailing.

**Comment by justinvyu** [comment]
@can-anyscale Has there already been a bisect for this?

**Comment by can-anyscale** [comment]
Yess, but bisect couldn't find anything;  could be some external changes break this

**Comment by justinvyu** [comment]
Recommendations given to solve this issue: https://github.com/facebookresearch/fairseq/issues/1529#issuecomment-567955547

**Comment by justinvyu** [comment]
Here's what I tried:
- Setting a lower learning rate 1e-5, still saw the same error 🔴 
- Shuffling the dataset with a different seed (made the error happen quicker) 🔴 
- Switching from global shuffle that happens once, to a local shuffle that will give different orderings per epoch. (This doesn't really affect anything due to the script only running for 1 epoch.) 🟡 
- Reducing the batch size to just use fp32. This works, but significantly slows down the training. (~35 minutes -> 3+ hours projected) 🟡 
  - Using gradient accumulation to still get the same effective batch size.
- Next idea to validate: try to set the `fp16-scale-tolerance=0.25` parameter so that we skip more batches before decreasing loss scale (less training happens, but it's ok just for this example to run successfully).

**Comment by justinvyu** [comment]
Setting the deepspeed fp16 configs `hysteresis=4` and `consecutive_hysteresis=True` basically gave me the same behavior as `fp16-scale-tolerance=0.25`.

You must have 4 batches in a row give NaN/inf gradients before reducing the loss scale factor by 2. Any non-NaN/inf gradient batch will reset the counter back to 4.

**Comment by can-anyscale** [comment]
Test passed on latest run: https://buildkite.com/ray-project/release/builds/4449#018c723c-8faa-4dd1-abe2-77fe9d3f9e6f
